commit 7d45f9220e55f29b2b4aa5fededa462af6718e80
Author: Volker Krause <vkrause@kde.org>
Date:   Sat May 13 13:04:05 2023 +0200

    Fix dependency finding in the CMake config file
    
    We need to find all public link dependencies here when building a shared
    library, but also all private link dependencies when building a static
    library.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 750ed771..5a6dd741 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -322,7 +322,6 @@ target_link_libraries(${PROJECT_NAME} PUBLIC ${Qt}::Core ${Qt}::Network ${Qt}::G
 if (Qt STREQUAL Qt5) # See #483
     target_link_libraries(${PROJECT_NAME} PRIVATE ${Qt}::Multimedia)
 endif()
-set(FIND_DEPS "find_dependency(${Qt}Keychain)") # For QuotientConfig.cmake.in
 
 if (${PROJECT_NAME}_ENABLE_E2EE)
     target_link_libraries(${PROJECT_NAME}
@@ -332,10 +331,6 @@ if (${PROJECT_NAME}_ENABLE_E2EE)
         PRIVATE
             OpenSSL::Crypto
     )
-    set(FIND_DEPS "${FIND_DEPS}
-    find_dependency(Olm)
-    find_dependency(OpenSSL)
-    find_dependency(${Qt}Sql)")
 endif()
 
 configure_file(${PROJECT_NAME}.pc.in ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc @ONLY NEWLINE_STYLE UNIX)
diff --git a/cmake/QuotientConfig.cmake.in b/cmake/QuotientConfig.cmake.in
index 1ddd0e20..fba513d3 100644
--- a/cmake/QuotientConfig.cmake.in
+++ b/cmake/QuotientConfig.cmake.in
@@ -1,6 +1,17 @@
 include(CMakeFindDependencyMacro)
 
-@FIND_DEPS@
+find_dependency(@Qt@Gui)
+find_dependency(@Qt@Network)
+find_dependency(@Qt@Keychain)
+if (@Quotient_ENABLE_E2EE@)
+    find_dependency(Olm)
+    find_dependency(OpenSSL)
+    find_dependency(@Qt@Sql)
+endif()
+
+if (NOT @BUILD_SHARED_LIBS@ AND NOT @BUILD_WITH_QT6@)
+    find_dependency(@Qt@Multimedia)
+endif()
 
 include("${CMAKE_CURRENT_LIST_DIR}/QuotientTargets.cmake")
 
